src/hotspot/share/opto/predicates.hpp

```diff
@@ -392,6 +394,30 @@ class TemplateAssertionPredicateExpressionNode : public StackObj {
   }
 };
 
+// This class creates a new Initialized Assertion Predicate.
+class InitializedAssertionPredicate : public StackObj {
+  IfNode* const _template_assertion_predicate;
+  NOT_PRODUCT(const AssertionPredicateType _assertion_predicate_type;)
```

Comment:
I don't see the value of this field. You might as well call `template_assertion_predicate->assertion_predicate_type()` at every use-site. There are only 2.
The issue with having a separate field here is that one wonders if this is really the same type as the one from the predicate above.

Reply:
Yes, they should have the same type. Since there is currently only one method which use this, I defined it there and removed the field for now. Eventually, we need to pass the type in since I want to have a single `TemplateAssertionPredicateNode` for the init and last value. But I can still reintroduce the field again then if needed.